### ISSUE TITLE
custom configuration support

### ISSUE THREAD

**Issue by bhcopeland** [kernelci/tuxmake#8]
***In GitLab by @terceiro on Jun 12, 2020, 19:31 GMT+1:***

must support both "make fooconfig" and passing arbitrary pre-made configuration files; must call olddefconfig in the later case anyway

**Assignees:** terceiro

*Migrated from GitLab: https://gitlab.com/Linaro/tuxmake/-/issues/8*

**Comment by bhcopeland** [comment]
***In GitLab by @terceiro on Jun 19, 2020, 21:14 GMT+1:***

this is already implemented, but only partially; the `olddefconfig` call is not implemented.

**Comment by bhcopeland** [comment]
***In GitLab by @terceiro on Jun 22, 2020, 18:08 GMT+1:***

tuxbuild config spec:https://gitlab.com/Linaro/kbapi/kbapi-documentation/-/blob/master/config.md

sample build log:

```
#
# make -sk KBUILD_BUILD_USER=TuxBuild -C/linux ARCH=x86 HOSTCC=gcc CC="sccache gcc" O=build defconfig
#
#
# 
set -e
cd /linux
export ARCH=x86
export HOSTCC=gcc
export CC=gcc
export CROSS_COMPILE=
scripts/kconfig/merge_config.sh -O build 'build/.config' 'build/frag.config' 

#
#
# make -sk KBUILD_BUILD_USER=TuxBuild -C/linux -j16 ARCH=x86 HOSTCC=gcc CC="sccache gcc" O=build
#
kernel/bpf/core.o: warning: objtool: ___bpf_prog_run()+0x22: call without frame pointer save/setup
#
# make -sk KBUILD_BUILD_USER=TuxBuild -C/linux -j16 ARCH=x86 HOSTCC=gcc CC="sccache gcc" O=build modules
#
#
# make -sk KBUILD_BUILD_USER=TuxBuild INSTALL_MOD_PATH=/linux/build/_modules_ INSTALL_MOD_STRIP=1 STRIP=strip -C/linux ARCH=x86 HOSTCC=gcc CC="sccache gcc" O=build modules_install
#
```

tuxbuild implementation: https://gitlab.com/Linaro/kbapi/kbapi/-/blob/master/builder/kernelci_build.py

**Comment by bhcopeland** [comment]
***In GitLab by @terceiro on Jul 3, 2020, 22:31 GMT+1:***

I have started on an implementation, and have a few questions: 

- do we keep treating the first kconfig as special?
- the approach exemplified in the tuxbuild spec writes to the source tree; so far tuxmake treats the source tree as readonly
  - we can emulate that by calling scripts/kconfig/merge_config.sh directly but that is maybe entering too much in implementation details of the Linux build system.
- KCONFIG_ALLCONFIG seems to be only used for randconfig; is that correct?
  - if so do we really need it in tuxmake, or can we just allow users to pass arbitrary environment variables?

**Comment by bhcopeland** [comment]
***In GitLab by @terceiro on Jul 8, 2020, 21:15 GMT+1:***

How to handle the initial config vs the extra ones

- Separate arguments kconfig + kconfig-add
- kconfig defaults to defconfig
- kconfig-add appends to the initial config with merge_config.sh

**Comment by bhcopeland** [comment]
***In GitLab by @terceiro on Jul 9, 2020, 18:27 GMT+1:***

mentioned in commit ae2a3c8fee77c1c0712b25f3cce931fd8ac85c4a

**Comment by bhcopeland** [comment]
***In GitLab by @terceiro on Jul 9, 2020, 21:11 GMT+1:***

closed via commit ae2a3c8fee77c1c0712b25f3cce931fd8ac85c4a

**Comment by bhcopeland** [comment]
***In GitLab by @terceiro on Jul 9, 2020, 21:11 GMT+1:***

mentioned in commit ea65c417c9f08e241308305dde46f584f9015540
